apps/server/src/controllers/userControllers/userUpdateKyc.ts

```diff
@@ -0,0 +1,90 @@
+import { Response } from 'express'
+import { RequestExt } from 'middleware/authorization/authentication'
+import { editProfileSchema } from '../../utils/validators'
+import { HTTP_STATUS_CODE } from '../../constants'
+import { v2 as cloudinary } from 'cloudinary'
+
+export const updateKycProfile = async (req: RequestExt, res: Response) => {
+  try {
+    const { _user: user, ...rest } = req.body
+
+    const validationResult = editProfileSchema.safeParse(rest)
+
+    if (!validationResult.success) {
+      return res.status(HTTP_STATUS_CODE.BAD_REQUEST).json({
+        message: 'Validation error',
+        issues: validationResult.error.issues,
+      })
+    }
+
+    const {
+      firstName,
+      lastName,
+      email,
+      phone,
+      gender,
+      occupation,
+      date_of_birth,
+      bvn,
+      address,
+      identification_type,
+      identification_number,
+      identification_doc,
+      proof_of_address_doc,
+    } = validationResult.data
+
+    const files = req.files as {
+      [fieldname: string]: Express.Multer.File[]
+    }
+    console.log(files)
+
+    let _identification_doc = identification_doc ?? user.identification_doc
+    let _proof_of_address_doc =
+      proof_of_address_doc ?? user.proof_of_address_doc
+
+    // upload files to Cloudinary if available.
+    const identificationDocPath = files?.['identification_doc']?.[0]?.path
+    const proofOfAddressDocPath = files?.['proof_of_address_doc']?.[0]?.path
+
+    if (identificationDocPath) {
+      const identificationDocResult = await cloudinary.uploader.upload(
+        files['identification_doc'][0].buffer.toString('base64')
+      )
+      _identification_doc = identificationDocResult.secure_url
+    }
+
+    if (proofOfAddressDocPath) {
+      const proofOfAddressDocResult = await cloudinary.uploader.upload(
+        files['proof_of_address_doc'][0].buffer.toString('base64')
+      )
+      _proof_of_address_doc = proofOfAddressDocResult.secure_url
+    }
```

Comment:
I'm thinking we could use a logical && operator to handle both conditions in the if blocks??

Reply:
I thought of that but what if the user wants to update only one document and not the two?🤷🏻